extensions/src/test/java/org/pentaho/platform/plugin/services/email/EmailServiceTest.java

```diff
@@ -0,0 +1,479 @@
+
+    emailConfigOriginal.setDefaultFrom( "from" );
+    emailConfigOriginal.setRefreshToken( "" );
+    assertEquals( false, emailService.isValid( emailConfigOriginal ) );
+
+  }
+
+  @Test
+  public void testIsValidEmailConfigAuthCode() {
+
+    final EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "test@pentaho.com", "Pentaho Scheduler", "", 36,
+                    EmailConstants.PROTOCOL_SMTP, true, "user", "password", false, true );
+    emailConfigOriginal.setAuthMechanism( EmailConstants.AUTH_TYPE_XOAUTH2 );
+    emailConfigOriginal.setClientId( "cid" );
+    emailConfigOriginal.setClientSecret( "clientsecret" );
+    emailConfigOriginal.setAuthorizationCode( "code" );
+    emailConfigOriginal.setGrantType( EmailConstants.GRANT_TYPE_AUTH_CODE );
+    emailConfigOriginal.setRedirectUri( "uri" );
+    boolean response = emailService.isValid( emailConfigOriginal );
+    assertEquals( true, response );
+
+    emailConfigOriginal.setClientId( "" );
+    assertEquals( false, emailService.isValid( emailConfigOriginal ) );
+
+    emailConfigOriginal.setClientId( "cid" );
+    emailConfigOriginal.setClientSecret( "" );
+    assertEquals( false, emailService.isValid( emailConfigOriginal ) );
+
+    emailConfigOriginal.setClientSecret( "secret" );
+    emailConfigOriginal.setDefaultFrom( "" );
+    assertEquals( false, emailService.isValid( emailConfigOriginal ) );
+
+    emailConfigOriginal.setDefaultFrom( "from" );
+    emailConfigOriginal.setAuthorizationCode( "" );
+    assertEquals( false, emailService.isValid( emailConfigOriginal ) );
+
+    emailConfigOriginal.setAuthorizationCode( "auth_code" );
+    emailConfigOriginal.setRedirectUri( "" );
+    assertEquals( false, emailService.isValid( emailConfigOriginal ) );
+  }
+
+  @Test
+  public void testIsValidEmailConfigNoAuth() {
+
+    final EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( false, false, "test@pentaho.com", "", "smtp.com", 36,
+                    EmailConstants.PROTOCOL_SMTP, true, "", "", false, true );
+    boolean response = emailService.isValid( emailConfigOriginal );
+    assertEquals( true, response );
+  }
+
+  @Test
+  public void testInvalidEmailConfig() {
+
+    final EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "", "", "smtp.com", 36,
+                    EmailConstants.PROTOCOL_SMTP, true, "", "", false, true );
+
+    boolean response = emailService.isValid( emailConfigOriginal );
+    assertEquals( false, response );
+  }
+
+  @Test
+  public void testSendEmailTestNoAuth() throws Exception {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( false, false, "test@pentaho.com", "Pentaho Scheduler", "", 25,
+                    EmailConstants.PROTOCOL_SMTP, true, "", "", true, true );
+    assertEquals( EmailService.TEST_EMAIL_SUCCESS, emailService.sendEmailTest( emailConfigOriginal ) );
```

Comment:
need to assert something about the MimeMessage being sent See MockMail and Scheuder/../EmailerTest.java

https://github.com/pentaho/pentaho-platform/blob/b572b4a8cc604219147fdadab7e70d4f274e684f/scheduler/src/test/java/org/pentaho/platform/scheduler2/email/EmailerTest.java#L76-L97